pkg/services/ngalert/schedule/registry.go

```diff
@@ -169,16 +169,19 @@ func (r *alertRulesRegistry) get(k models.AlertRuleKey) *models.AlertRule {
 	return r.rules[k]
 }
 
-// set replaces all rules in the registry.
-func (r *alertRulesRegistry) set(rules []*models.AlertRule, folders map[string]string) {
+// set replaces all rules in the registry. Returns difference between previous and the new current version of the registry
```

Comment:
I'm not sure I understand the purpose of this code as don't we check the version when reading from `updateCh`?

Reply:
This is the last part of the solution. Please refer to PR description

Reply:
we do, but the scheduler needs to know whether the version has changed or not just to not call updateCh every tick. Scheduler does not have access to version that is held in ruleEvalRoutine context